internal/db/db.go

```diff
@@ -143,115 +144,253 @@ func (tx *holdTx) FindOne(modelID uuid.UUID, matcher Matcher) (rec Record, err e
 	return
 }
 
-func (tx holdTx) FindMany(modelID uuid.UUID, matcher Matcher) []Record {
-	mi := tx.h.IterMatches(modelID, matcher)
-	hits := []Record{}
-	for val, ok := mi.Next(); ok; val, ok = mi.Next() {
-		hits = append(hits, val)
+func (tx *holdTx) FindMany(modelID uuid.UUID, matcher Matcher) (recs []Record, err error) {
+	recs, err = tx.h.FindMany(modelID, matcher)
+	return
+}
+
+func (tx *holdTx) Insert(rec Record) error {
+	tx.ensureWrite()
+	h, err := tx.h.Insert(rec)
+	if err != nil {
+		return err
 	}
-	return hits
+	tx.h = h
+	return nil
 }
 
-func (tx *holdTx) Insert(rec Record) {
+func (tx *holdTx) Update(r Record, fields map[string]interface{}) (Record, error) {
```

Comment:
I'm not sure if the interface at the DB level should be a map[string]interface{}.. that's leaking the JSON-ness of the API into the DB a bit. I wonder if the API should just be
`Update(id uuid.UUID, newRecord db.Record)`

Reply:
Why do you need the id? the record should already have it